sktime/forecasting/naive_variance.py

```diff
@@ -0,0 +1,192 @@
+
+        Users can also implement _predict_interval if calling it makes this faster.
+
+        Parameters
+        ----------
+        fh : int, list, np.array or ForecastingHorizon
+            Forecasting horizon
+        X : pd.DataFrame, optional (default=None)
+            Exogenous time series
+        alpha : float or list of float, optional (default=0.5)
+            A probability or list of, at which quantile forecasts are computed.
+
+        Returns
+        -------
+        pred_quantiles : pd.DataFrame
+            Column has multi-index: first level is variable name from y in fit,
+                second level being the quantile forecasts for each alpha.
+                Quantile forecasts are calculated for each a in alpha.
+            Row index is fh. Entries are quantile forecasts, for var in col index,
+                at quantile probability in second-level col index, for each row index.
+        """
+        self.check_is_fitted()
+        alpha = check_alpha(alpha)
+
+        pred_var = self.predict_var(fh, X)
+
+        errors = []
+        for a in alpha:
+            z = _zscore(1 - a)
+            errors.append(pred_var * z)
+
+        y_pred = self.predict(fh, X)
+
+        pred_quantiles = pd.DataFrame()
+        for a, error in zip(alpha, errors):
+            if a < 0.5:
+                pred_quantiles[a] = y_pred - error
+            else:
+                pred_quantiles[a] = y_pred + error
+
+        arrays = [len(alpha) * ["Quantiles"], alpha]
+        columns = pd.MultiIndex.from_tuples(list(zip(*arrays)))
+        pred_quantiles.columns = columns
+
+        return pred_quantiles
+
+    def _predict_var(self, fh, X=None, cov=False):
+        """
+        Compute/return prediction variance for a forecast.
+
+        Must be run *after* the forecaster has been fitted.
+
+        Parameters
+        ----------
+        fh : int, list, np.array or ForecastingHorizon
+            Forecasting horizon
+        X : pd.DataFrame, optional (default=None)
+            Exogenous time series
+        cov : bool, optional (default=False)
+            Wether to return marginal variances or covariance matrices.
+
+        Returns
+        -------
+        pred_var : np.ndarray
+            if cov=False,
+                a vector of same length as fh with predictive marginal variances;
+            if cov=True,
+                a square matrix of size len(fh) with predictive covariance matrix.
+        """
+        self.forecaster.check_is_fitted()
+
+        fh_ = check_fh(fh)
+
+        n_timepoints = len(self._y)
+        residuals = []
+        for i in range(0, n_timepoints):
+            # a duplicate for rolling window fit/predict
+            forecaster = clone(self.forecaster)
+
+            try:
+                forecaster.fit(self._y[: i + 1])
+                y_hat = forecaster.predict(self._y.index, self._X)
+                residuals.append(y_hat - self._y)
```

Comment:
alternatively, you can use `predict_residuals`

Reply:
Just found how to use it, nice shortcut